### PR TITLE
Make lightning gateway optional

### PR DESCRIPTION
**Pull request by justinmoon**
Builds on #174. Just look at 2nd commit.

Our README on master right now claims that the gateway is optional, but our code doesn't. Instead, we just [generate a random node pubkey](https://github.com/fedimint/minimint/blob/master/ln-gateway/src/bin/gw_configgen.rs#L27)! But this isn't a problem because it would only be used in an invoice and we haven't done that, yet.

So this PR attempt to make the gateway optional. It also stops having the gateway code generate `client.json` and instead adds a new command to `configgen` to do this. Along the way I added a `/info` endpoint to the gateway HTTP server so you can look up gateway public keys.

### CODE REVIEW

File: client/client-lib/src/lib.rs [hunk 1]
```diff
@@ -46,5 +46,5 @@ impl<CO> OwnedClientContext<CO> {
 #[derive(Debug, Serialize, Deserialize)]
 pub struct ClientAndGatewayConfig {
     pub client: ClientConfig,
-    pub gateway: LightningGateway,
+    pub gateway: Option<LightningGateway>,
```
Comment (justinmoon) [inline client/client-lib/src/lib.rs:49]: This should be a `Vec`

File: README.md [hunk 2]
```diff
@@ -114,8 +101,8 @@ Send your gateway Ligtning node some funds:
 # Get address from gateway lightning node (we'll call it LN_ADDRESS)
 lightning-cli --network regtest --lightning-dir=ln1 newaddr
 
-# Send bitcoin to gateway lightning address
-bitcoin-cli -regtest -rpcuser=bitcoin -rpcpassword=bitcoin sendtoaddress <LN_ADDRESS> <AMOUNT>
+# Send bitcoin to gateway lightning address, and mine a block to confirm it
+bitcoin-cli -regtest -rpcuser=bitcoin -rpcpassword=bitcoin sendtoaddress <LN_ADDRESS> 1.0
```
Comment (elsirion) [inline README.md:105]: Should have large enough denominations for this to not take days (e.g. if the server had to issue 100'000 tokens)

**Review by elsirion — COMMENTED** [review]
Cool! Just noticed the dependency on the GW plugin, hope to merge soon.

**Comment by justinmoon** [comment]
Closing in favor of #269

**Pull request closed without merging**